tests/components/energy/test_sensor.py

```diff
@@ -78,19 +101,35 @@ async def test_cost_sensor_price_entity(hass, hass_storage) -> None:
         "data": energy_data,
     }
 
+    now = dt_util.utcnow()
     last_reset = dt_util.utc_from_timestamp(0).isoformat()
-    hass.states.async_set(
-        "sensor.energy_consumption",
-        "0",
-        {"last_reset": last_reset},
-    )
-    hass.states.async_set("sensor.energy_price", "1")
 
-    now = dt_util.utcnow()
+    # Optionally initialize dependent entities
+    if initial_energy is not None:
+        hass.states.async_set(
+            "sensor.energy_consumption", initial_energy, {"last_reset": last_reset}
+        )
+    hass.states.async_set("sensor.energy_price", "1")
 
     with patch("homeassistant.util.dt.utcnow", return_value=now):
         await setup_integration(hass)
 
+    state = hass.states.get("sensor.energy_consumption_cost")
+    assert state.state == initial_cost
+    assert state.attributes[ATTR_DEVICE_CLASS] == DEVICE_CLASS_MONETARY
+    if initial_cost != "unknown":
+        assert state.attributes[ATTR_LAST_RESET] == now.isoformat()
+    assert state.attributes[ATTR_STATE_CLASS] == STATE_CLASS_MEASUREMENT
+    assert state.attributes[ATTR_UNIT_OF_MEASUREMENT] == "€"
+
+    # Optional late setup of dependent entities
+    if initial_energy is None:
```

Comment:
If we can't solve these conditions I'd split the test into separate tests.

Reply:
90% of the test is shared between the two cases. We would need to keep them in sync if we split it out. I think both ways are fine.

Reply:
What's the ratio of time spent reading vs writing code? What's the cost of a bug vs the cost of maintaining an extra test? I'd say avoiding conditions in tests will make them easier to read and less error prone.

Reply:
We try all the different types of configuration. I started doing this, but it ends up being 8 copies of the same test with 2-3 lines difference. I disagree that it will improve readability in this case.

Reply:
The only way splitting up can become more readable if we decide to not fully test all features for each permutation of the configuration, but that doesn't make sense.

Reply:
Ok. I'll take another look at this later and see if there's something we can do to improve it.